gourmet/gtk_extras/dialog_extras.py

```diff
@@ -1056,7 +1058,7 @@ def is_extension_legal (self, fn):
             if fnmatch.fnmatch(fn, e):
                 return True
 
-    def run (self):
+    def run(self) -> Union[str, List[str], None]:
```

Comment:
Thank you for that annotation. Looking at that type, I think we should try to return either None or a list. So the `str` return we would return as a singleton list. This would reduce the type to `Optional[List[str]]`. But `List` already encodes optional via the empty list.

Would there be anything wrong with returning always `List[str]`? A single value is the same as iterating over a singleton list, `bool()` has the same behaviour, etc.

Reply:
Like so? :)

Reply:
Yeah. I think this is a consistent wrap around that functionality. Great work!